packages/browser-integration-tests/suites/replay/bufferMode/test.ts

```diff
@@ -0,0 +1,300 @@
+import {
+  getReplayEvent,
+  getReplayRecordingContent,
+  isReplayEvent,
+  shouldSkipReplayTest,
+  waitForReplayRequest,
+} from '../../../utils/replayHelpers';
+
+sentryTest(
+  '[buffer-mode] manually start buffer mode and capture buffer',
+  async ({ getLocalTestPath, page, browserName }) => {
+    // This was sometimes flaky on firefox/webkit, so skipping for now
+    if (shouldSkipReplayTest() || ['firefox', 'webkit'].includes(browserName)) {
+      sentryTest.skip();
+    }
+
+    let callsToSentry = 0;
+    let errorEventId: string | undefined;
+    const reqPromise0 = waitForReplayRequest(page, 0);
+    const reqPromise1 = waitForReplayRequest(page, 1);
+    const reqPromise2 = waitForReplayRequest(page, 2);
+    const reqErrorPromise = waitForErrorRequest(page);
+
+    await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+      const event = envelopeRequestParser(route.request());
+      // error events have no type field
+      if (event && !event.type && event.event_id) {
+        errorEventId = event.event_id;
+      }
+      // We only want to count errors & replays here
+      if (event && (!event.type || isReplayEvent(event))) {
+        callsToSentry++;
+      }
+
+      return route.fulfill({
+        status: 200,
+        contentType: 'application/json',
+        body: JSON.stringify({ id: 'test-id' }),
+      });
+    });
+
+    const url = await getLocalTestPath({ testDir: __dirname });
+
+    await page.goto(url);
+    await page.click('#go-background');
+    await page.click('#error');
+    await new Promise(resolve => setTimeout(resolve, 1000));
+
+    // error, no replays
+    expect(callsToSentry).toEqual(1);
+    await reqErrorPromise;
+
+    expect(
+      await page.evaluate(() => {
+        const replayIntegration = (window as unknown as Window & { Replay: { _replay: ReplayContainer } }).Replay;
+        const replay = replayIntegration._replay;
+        return replay.isEnabled();
+      }),
+    ).toBe(false);
+
+    // Start buffering and assert that it is enabled
+    expect(
+      await page.evaluate(() => {
+        const replayIntegration = (window as unknown as Window & { Replay: InstanceType<typeof Replay> }).Replay;
+        // @ts-ignore private
+        const replay = replayIntegration._replay;
+        replayIntegration.startBuffering();
+        return replay.isEnabled();
+      }),
+    ).toBe(true);
+
+    await page.click('#log');
+    await page.click('#go-background');
+    await page.click('#error2');
+    await new Promise(resolve => setTimeout(resolve, 1000));
+
+    // 2 errors
+    expect(callsToSentry).toEqual(2);
+
+    await page.evaluate(async () => {
+      const replayIntegration = (window as unknown as Window & { Replay: Replay }).Replay;
+      await replayIntegration.flush();
+    });
+
+    const req0 = await reqPromise0;
+
+    // 2 errors, 1 flush
+    expect(callsToSentry).toEqual(3);
```

Comment:
This seems to be flaky: https://github.com/getsentry/sentry-javascript/actions/runs/4736874296/jobs/8408976370